### DISCUSSION TITLE
The ability to declaratively create SLO and alerting channels

### DISCUSSION THREAD

**Discussion by distinguished-sre** [coroot/coroot#204] [General]
It would be convenient to create an [SLO](https://coroot.com/docs/coroot-community-edition/inspections/slo) measurement for individual applications via the [Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) when their helm chart is deployed, or set up a [Alerting integrations](https://coroot.com/docs/coroot-community-edition/alerting/overview) when coroot itself is deployed

### Proposal

We require the ability to create resources in a decorative manner in the form of a [SLO](https://coroot.com/docs/coroot-community-edition/inspections/slo), or to connection to the [Alerting integrations](https://coroot.com/docs/coroot-community-edition/alerting/overview) for helm charts in the form of [Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)

**Comment by AmazingTurtle** [comment]
I would love to see this especially considering that for example a `redis` `master-replica` setup frequently violates `SLO` due to high latency caused by `REPLCONF ACK`

![image](https://github.com/user-attachments/assets/dac203a1-8113-4836-89b5-945103da27ff)
